x-pack/plugins/reporting/public/panel_actions/get_csv_panel_action.tsx

```diff
@@ -78,31 +76,20 @@ export class ReportingCsvPanelAction implements ActionDefinition<ActionContext>
   }
 
   public isCompatible = async (context: ActionContext) => {
-    await new Promise<void>((resolve) => {
-      this.startServices$.subscribe(([{ application }, { licensing }]) => {
-        licensing.license$.subscribe((license) => {
-          const results = license.check('reporting', 'basic');
-          const { showLinks } = checkLicense(results);
-          this.licenseHasDownloadCsv = showLinks;
-        });
-
-        if (this.usesUiCapabilities) {
-          this.capabilityHasDownloadCsv = application.capabilities.dashboard?.downloadCsv === true;
-        } else {
-          this.capabilityHasDownloadCsv = true; // deprecated
-        }
-
-        resolve();
-      });
-    });
+    const { embeddable } = context;
 
-    if (!this.licenseHasDownloadCsv || !this.capabilityHasDownloadCsv) {
+    if (embeddable.type !== 'search') {
       return false;
     }
 
-    const { embeddable } = context;
+    const [{ application }, { licensing }] = await firstValueFrom(this.startServices$);
```

Comment:
the logic is the same, I've just rewrote to remove the race condition with this.licenseHasDownloadCsv set too late.